File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/LMS/DiscussionRoom.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.LMS
{
    public class DiscussionRoom : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        [Required(ErrorMessage = "Title is required")]
        [StringLength(100, ErrorMessage = "Title cannot be longer than 100 characters.")]
        public string Title { get; set; }
        public string MeetingLink { get; set; }
        public string? RecordLink { get; set; }
        public string? ImageLink { get; set; }
        public bool IsRepeatedWeekly { get; set; }
        public DateOnly StartDate { get; set; }
        public DateOnly EndDate { get; set; }
        public string Time { get; set; }

        public bool Saturday { get; set; }
        public bool Sunday { get; set; }
        public bool Monday { get; set; }
        public bool Tuesday { get; set; }
        public bool Wednesday { get; set; }
        public bool Thursday { get; set; }
        public bool Friday { get; set; } 

        [ForeignKey("School")]
        public long SchoolID { get; set; }
        public School School { get; set; }
        public ICollection<DiscussionRoomStudentClassroom> DiscussionRoomStudentClassrooms { get; set; } = new List<DiscussionRoomStudentClassroom>();
    }
}

File: LMS_CMS_Frontend/src/app/Models/LMS/discussion-room.ts
import { DiscussionRoomStudentClassroom } from "./discussion-room-student-classroom";

export class DiscussionRoom {
    constructor(
        public id: number = 0,
        public schoolID: number = 0,
        public title: string = '',
        public imageLink: string = '',
        public meetingLink: string = '',
        public recordLink: string = '',
        public startDate: string = '',
        public endDate: string = '',
        public time: string = '',
        public isRepeatedWeekly: boolean = false,
        public saturday: boolean = false,
        public sunday: boolean = false,
        public monday: boolean = false,
        public tuesday: boolean = false,
        public wednesday: boolean = false,
        public thursday: boolean = false,
        public friday: boolean = false,
        public insertedByUserId :number =0,
        public discussionRoomStudentClassrooms : DiscussionRoomStudentClassroom[]=[],
        public studentClassrooms : number[]=[],
        public imageFile : File|null = null
    ) {}
}   